forem/forem
# Add Recent Vomit Reaction Reports to Users' Admin Users Page

## What type of PR is this? (check all applicable)


## Description
This PR adds a "Recent Reactions" section to a User's Admin User page (`/admin/users/:id/edit`). The reactions section displays a User's most recent `Reactions` with useful data (`path`, `reactable_type`, `title`, `category`, and `created_at`). By adding this data to the User's Admin User page, Admins now have the ability to more easily view which reactions that a User has and any related detail for the reactions, allowing Admins to refer back to this data if need be.

## Related Tickets & Documents
Relates to a private issue and relates to PR #10058 

## QA Instructions, Screenshots, Recordings
**To QA this PR, first ensure that you have seeded your DB so that you have `vomit` reactions (if you don't have any reactions seeded, `vomit` on some content!) and have `super_admin` permissions, then please do the following:**
- Navigate to `/admin/users/:id/edit` in your browser:
<img width="1388" alt="Screen Shot 2020-08-28 at 11 25 29 AM" src="https://user-images.githubusercontent.com/32834804/91595921-90c14800-e921-11ea-817f-b349ab1b44da.png">

- Click the "Toggle" button next to "Recent Reactions" to see a list of all of the `vomit` reactions for the particular user:
<img width="1355" alt="Screen Shot 2020-08-28 at 2 40 23 PM" src="https://user-images.githubusercontent.com/32834804/91613336-8e201c00-e93c-11ea-9920-7cfead8e6065.png">

- If the user has been vomited on, the reaction will display as shown below:
<img width="1158" alt="Screen Shot 2020-08-28 at 2 40 37 PM" src="https://user-images.githubusercontent.com/32834804/91613415-b60f7f80-e93c-11ea-8d73-61b9cb7095cf.png">

- If the user does not have any `vomit` reactions, this message should be displayed:
<img width="1127" alt="Screen Shot 2020-08-28 at 2 00 49 PM" src="https://user-images.githubusercontent.com/32834804/91610801-2adfbb00-e937-11ea-99af-0faf2a79ed1a.png">

- Click the `vomit` reaction to navigate to content related to the `vomit` reaction:
<img width="1327" alt="Screen Shot 2020-08-28 at 11 26 08 AM" src="https://user-images.githubusercontent.com/32834804/91595961-a46cae80-e921-11ea-994e-b06eff7a5587.png">

- Try to break things! 🛠️ 

## Added tests?


## Added to documentation?


## [optional] Are there any post-deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

![Dog in the office](https://media.giphy.com/media/10mbtDA1OCEgDK/giphy.gif)
